### ISSUE TITLE
ws::client-handler-data

### ISSUE THREAD

**Issue by mishoo** [3b/clws#12]
The `:documentation` bit on [this slot](https://github.com/3b/clws/blob/fcaecfdbb06e3f065d10592f977a5aed5b4de595/client.lisp#L60) implies that I can store whatever I need into it, but the fact it's not exported makes me a bit nervous.  Am I using it properly?  Shouldn't it be in the exports?

A little suggestion for improvement would be to provide an option — like hunchensocket — to name the class to be instantiated for clients.  That'd be a `client-class` slot on the `ws-resource` class, which is exported and we're encouraged to inherit from it.  That would allow users to add whatever slots they need.  I'd be happy to work on it and send a PR if it stands chances to be approved.

Just for the record, one thing I tried was to use `(change-class client 'my-own-class)` in my specialization of `ws:resource-client-connected`.  That works nicely, about 80% of the time :-)  But [randomly](http://ircbrowse.net/day/lisp/today?id=9487714&timestamp=1455118150#t1455118150) it'll lead to a "invalid frame header" error.  That's likely a bug in SBCL; I'd report it, but I didn't yet manage to break it down to a simple test case...

**Comment by 3b** [comment]
Yeah, it does look like `client-handler-data` should be exported.

I'm not sure if things are created in the right order for the resource to specify the client class, but it seems like a reasonable thing to do if it does work.

I suspect the problem with `change-class` in `resource-client-connected` is threading, possibly it would work better from `resource-accept-connection`? Not sure if it is a good idea or not either way.

PR for either option would probably be accepted, mostly just depends on how long it takes me to figure out what the patch is doing (or the original code for that matter).

**Comment by erjoalgo** [comment]
Following up on this thread. Is this the recommended way of associating client-specific metadata?
